code-engine/main.tf

```diff
@@ -52,7 +52,10 @@ locals {
   cd_code_engine_project_prefix = (var.cd_code_engine_project_prefix == "") ? var.code_engine_project_prefix : var.cd_code_engine_project_prefix
   ci_code_engine_project        = (var.ci_code_engine_project == "") ? var.code_engine_project : var.ci_code_engine_project
   cd_code_engine_project        = (var.cd_code_engine_project == "") ? var.code_engine_project : var.cd_code_engine_project
-  registry_namespace            = format("%s%s", var.registry_namespace, var.registry_namespace_suffix)
+
+  ci_code_engine_project_name = (local.ci_code_engine_project_prefix == "") ? local.ci_code_engine_project : format("%s-%s", local.ci_code_engine_project_prefix, local.ci_code_engine_project)
+  cd_code_engine_project_name = (local.cd_code_engine_project_prefix == "") ? local.cd_code_engine_project : format("%s-%s", local.cd_code_engine_project_prefix, local.cd_code_engine_project)
+  registry_namespace          = format("%s%s", var.registry_namespace, var.registry_namespace_suffix)
```

Comment:
would it be possible to also add a random suffix to the namespace name. The namespace name must be globally unique (across all accounts), and users have been reporting clashes in their testing. This is also done for cos buckets which have the same characteristics ( https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/521ced30f57624f87a8f2465e7348fbfcaa7c690/main.tf#L116 )